Common/itkTransformixInputPointFileReader.hxx

```diff
@@ -40,30 +40,60 @@ TransformixInputPointFileReader<TOutputMesh>::GenerateOutputInformation()
   {
     this->m_Reader.close();
   }
-  this->m_Reader.open(this->m_FileName);
 
-  /** Read the first entry */
-  std::string indexOrPoint;
-  this->m_Reader >> indexOrPoint;
+  if( this->m_Binary ) {
+    this->m_Reader.open(this->m_FileName, std::ios::binary);
+
+    /** Read index flag */
+    unsigned long isindex;
+    this->m_Reader.read((char*) &isindex, sizeof(isindex));
```

Comment:
Instead of C-style `(char*)` cast, please use `reinterpret_cast<const char *>`. C-style casts are generally considered unsafe.